### PR TITLE
Falling back to public EnsEMBL if specified unavailable

### PR DESCRIPTION
**Pull request by kayrein**

### CODE REVIEW

File: lib/WebAppCommon/Util/EnsEMBL.pm [hunk 1]
```diff
@@ -20,10 +21,21 @@ class_has registry => (
 
 sub _build_registry {
 
-    Bio::EnsEMBL::Registry->load_registry_from_db(
-        -host => $ENV{LIMS2_ENSEMBL_HOST} || 'ensembldb.internal.sanger.ac.uk',
-        -user => $ENV{LIMS2_ENSEMBL_USER} || 'anonymous',
-    );
+    my $public = 'ensembldb.ensembl.org';
+    my $host   = $ENV{LIMS2_ENSEMBL_HOST};
```
Comment (Jana-A) [inline lib/WebAppCommon/Util/EnsEMBL.pm:25]: what if LIMS2_ENSEMBL_HOST is not set but the Ensembl mirror is working?

File: lib/WebAppCommon/Util/EnsEMBL.pm [hunk 2]
```diff
@@ -20,10 +21,21 @@ class_has registry => (
...
+    my $loaded;
+    try {
+        $loaded = Bio::EnsEMBL::Registry->load_registry_from_db(
+            -host => $host || $public,
+            -user => $ENV{LIMS2_ENSEMBL_USER} || 'anonymous',
+        );
+    };
+    if ( not $loaded and $host ) {
```
Comment (Jana-A) [inline lib/WebAppCommon/Util/EnsEMBL.pm]: You can also just catch the error from trying to load public and internal Ensembl.

File: shared_static/js/genoverse_custom_tracks.js [hunk 3]
```diff
@@ -1,25 +1,24 @@
...
-
 function detectIE() {
-        var ua = window.navigator.userAgent;
-
-        var msie = ua.indexOf('MSIE ');
-        if (msie > 0) {
-            // IE 10 or older => return version number
-            return parseInt(ua.substring(msie + 5, ua.indexOf('.', msie)), 10);
-        }
+    var ua = window.navigator.userAgent;
```
Comment (Jana-A) [inline shared_static/js/genoverse_custom_tracks.js]: Is this also related to Ensembl?

File: lib/WebAppCommon/Util/EnsEMBL.pm (hunk 2 above)
Comment (kayrein) [reply lib/WebAppCommon/Util/EnsEMBL.pm]: Unfortunately that doesn't work, for some error conditions the EnsEMBL API throws an exception and for some others it just returns 0. By wrapping it in the _try_ we prevent the exception from bubbling up, so either way we wind up with a false value in $loaded if EnsEMBL couldn't be loaded.

File: lib/WebAppCommon/Util/EnsEMBL.pm (hunk 1 above)
Comment (kayrein) [reply lib/WebAppCommon/Util/EnsEMBL.pm:25]: That seems like the point of the LIMS2_ENSEMBL_HOST variable - if it's set, that's because there's a mirror we should try first. In the case of WGE, this may be run externally and ideally we won't have references to our internal resources in the code. If LIMS2 is consuming it, then in the production and staging environments the environment variables will be set.

**Comment by Jana-A** [comment]
This pull request is good to go.
